pandas/core/indexing.py

```diff
@@ -1822,10 +1855,21 @@ def _getitem_axis(self, key, axis: int):
                 indexer[axis] = locs
                 return self.obj.iloc[tuple(indexer)]
 
+        if self.regex and isinstance(key, str):
```

Comment:
would we want to raise if regex and key is not str?

Reply:
IMO no.

The .loc interface should anyway accept one axis selector that is not a string (e.g. ``df.loc(regex=True)[df.A ==1, "^col_"]``, so letting neither be a string would not be a big difference and just mean no effect from setting the regex parameter.

Reply:
I see both points but tend to agree more with @simonjayhawkins. At the very least wouldn't we want to accept other types (bytes, re.pater)?

Reply:
Yeah, it could accept bytes. What do you mean by re.pater?